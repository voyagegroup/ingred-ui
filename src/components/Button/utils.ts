import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
