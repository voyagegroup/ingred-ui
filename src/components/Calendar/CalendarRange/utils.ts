import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
