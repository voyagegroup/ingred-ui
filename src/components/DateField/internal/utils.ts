export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
