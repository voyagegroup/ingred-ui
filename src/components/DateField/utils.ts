import { useMemo } from "react";

type Sections = {
  start: number;
  end: number;
  value: string;
  editable: boolean;
};

/**
 * 何らかのフォーマットで入ってくる日付を開始位置と終了位置と値を持つセクションに分割する
 * useDateField で format された日付操作を汎用的に行うために必要なプロパティを返す
 * 例) 2023-01-02 -> [
 *   { start: 0, end: 3, value: "2023", editable: true },
 *   { start: 4, end: 4, value: "-", editable: false },
 *   { start: 5, end: 6, value: "01", editable: true },
 *   { start: 7, end: 7, value: "-", editable: false },
 *   { start: 8, end: 9, value: "02", editable: true }
 * ]
 *
 * @param formattedDate 何らかのフォーマットで入ってくる日付
 * @returns 開始位置と終了位置と値を持つセクション
 */
export const getSections = (formattedDate?: string | null) => {
  if (!formattedDate) {
    return [];
  }

  const sections: Sections[] = [];
  let start = 0;
  let isPrevCharDigit = !isNaN(Number(formattedDate[0]));

  for (let index = 1; index <= formattedDate.length; index++) {
    const currentChar = formattedDate[index];
    const isCurrentCharDigit =
      !isNaN(Number(currentChar)) && currentChar !== " ";

    if (
      isCurrentCharDigit !== isPrevCharDigit ||
      currentChar === " " ||
      index === formattedDate.length
    ) {
      sections.push({
        start,
        end: index - 1,
        value: formattedDate.slice(start, index),
        editable: isPrevCharDigit,
      });
      start = index;
      isPrevCharDigit = isCurrentCharDigit;
    }
  }

  return sections;
};

/**
 * 開始位置と終了位置と値を持つセクションをフォーマットされた日付に変換する
 */
export const formatString = (sectionsWithCharactor: Sections[]) =>
  sectionsWithCharactor.map((section) => section.value).join("");

type ReactRef<T> =
  | React.RefCallback<T>
  | React.MutableRefObject<T>
  | React.ForwardedRef<T>
  | string
  | null
  | undefined;

// from: https://github.com/voyagegroup/ingred-ui/blob/master/src/hooks/useMergeRefs.ts
export function useMergeRefs<T>(...refs: ReactRef<T>[]): React.Ref<T> {
  return useMemo(() => {
    if (refs.every((ref) => ref === null)) {
      return null;
    }
    return (refValue: T) => {
      for (const ref of refs) {
        if (typeof ref === "function") {
          ref(refValue);
        } else if (ref && typeof ref !== "string") {
          ref.current = refValue;
        }
      }
    };
  }, [refs]);
}
