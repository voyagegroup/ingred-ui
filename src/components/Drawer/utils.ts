/**
 * 相対値をpx値に変換するヘルパー関数
 * @param value - 変換する値（数値またはvw/vh/%/px等の単位付き文字列）
 * @param direction - Drawerの方向（変換時の基準値決定に使用）
 * @returns px値（数値）
 */
export const convertToPixels = (value: string | number, direction: "left" | "right" | "bottom"): number => {
  if (typeof value === "number") {
    return value; // 数値はpx扱い
  }
  
  // 文字列の場合は単位チェック
  const numericValue = parseFloat(value);
  
  if (value.includes("vw")) {
    return (numericValue / 100) * window.innerWidth;
  }
  if (value.includes("vh")) {
    return (numericValue / 100) * window.innerHeight;
  }
  if (value.includes("%")) {
    // %の場合は親要素基準だが、ここではviewport基準として扱う
    const isHorizontal = direction === "left" || direction === "right";
    return (numericValue / 100) * (isHorizontal ? window.innerWidth : window.innerHeight);
  }
  if (value.includes("px")) {
    return numericValue;
  }
  
  // 単位なしの文字列は警告を出す
  if (!isNaN(numericValue)) {
    console.warn(`Drawer: 文字列サイズには単位を指定してください。"${value}" → "${value}px" を推奨`);
    return numericValue; // 一応数値として扱う
  }
  
  // 無効な値の場合はエラー警告とフォールバック
  console.error(`Drawer: 無効なサイズ値です: "${value}". デフォルト値を使用します。`);
  return 400; // フォールバック
}; 