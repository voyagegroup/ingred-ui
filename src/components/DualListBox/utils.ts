import {
  CandidateItem,
  DualListBoxCandidateItem,
  DualListBoxItem,
} from "./DualListBox";

export const getCandidateItems = (
  candidateItems: DualListBoxCandidateItem[],
  selectedItems: DualListBoxItem[],
) => {
  const result: CandidateItem[] = [];
  candidateItems.forEach((candidateItem) => {
    if (candidateItem.items) {
      result.push({
        ...candidateItem,
        items: getCandidateItems(candidateItem.items, selectedItems),
      });
      return;
    }

    const targetItem = selectedItems.find(
      (selectedItem) => selectedItem.id === candidateItem.id,
    );
    if (targetItem) {
      result.push({
        ...candidateItem,
        selected: true,
      });
    } else {
      result.push(candidateItem);
    }
  });

  return result;
};
