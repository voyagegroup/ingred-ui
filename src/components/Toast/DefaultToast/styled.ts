import styled, { keyframes } from "styled-components";

export const Wrapper = styled.div<{ transitionDuration: number }>`
  padding: ${({ theme }) => theme.spacing * 2}px;
  transition: height ${({ transitionDuration }) => transitionDuration - 100}ms
    100ms;
`;

type ContainerProps = {
  background: string;
  boxShadow: string;
  transitionDuration: number;
  transform: string;
  opacity: number;
};
export const Container = styled.div<ContainerProps>`
  position: relative;
  display: flex;
  align-items: center;
  min-width: 400px;
  max-width: 800px;
  height: 64px;
  padding: 0 ${({ theme }) => theme.spacing * 2}px;
  background: ${({ background }) => background};
  box-shadow: ${({ boxShadow }) => boxShadow};
  transition: transform ${({ transitionDuration }) => transitionDuration}ms
      ease-in-out,
    opacity ${({ transitionDuration }) => transitionDuration}ms;
  transform: ${({ transform }) => transform};
  opacity: ${({ opacity }) => opacity};
  border-radius: ${({ theme }) => theme.radius}px;
  overflow: hidden;
`;

export const IconContainer = styled.div<{ background: string }>`
  flex-shrink: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 28px;
  height: 28px;
  margin-right: ${({ theme }) => theme.spacing * 2}px;
  background: ${({ background }) => background};
  border-radius: 50%;
`;

export const CloseIconContainer = styled.div`
  cursor: pointer;
  margin-left: ${({ theme }) => theme.spacing * 2}px;
`;

type CountdownProps = {
  autoDismissTimeout: number;
  opacity: number;
  isRunning: boolean;
  backgroundColor: string;
};
export const shrinkKeyframes = keyframes`
  from {
    width: 100%;
  }
  to {
    width: 0%
  }
`;
export const Countdown = styled.div<CountdownProps>`
  animation: ${shrinkKeyframes}
    ${({ autoDismissTimeout }) => autoDismissTimeout}ms linear;
  background-color: ${({ backgroundColor }) => backgroundColor};
  top: 0;
  left: 0;
  opacity: ${({ opacity }) => opacity};
  position: absolute;
  width: 0;
  height: 3px;
`;
