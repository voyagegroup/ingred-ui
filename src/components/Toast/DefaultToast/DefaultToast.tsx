import * as React from "react";
import * as Styled from "./styled";
import Flex from "../../Flex";
import Icon from "../../Icon";
import Typography from "../../Typography";
import { Theme, useTheme } from "../../../themes";
import { hexToRgba } from "../../../utils/hexToRgba";
import {
  AppearanceTypes,
  Placement,
} from "../../../lib/react-toast-notification/src/types";
import { ToastProps } from "../../../lib/react-toast-notification/src/ToastElement";

type ToastStyle = {
  background: string;
  boxShadow: string;
  icon: JSX.Element;
  iconBackground: string;
  countDownBackground: string;
  titleColor: string;
  contentColor: string;
  closeColor: string;
};

const getToastStyles = ({
  palette,
}: Theme): Record<AppearanceTypes, ToastStyle> => ({
  info: {
    background: palette.primary.highlight,
    boxShadow: `0px 0px 16px ${hexToRgba(palette.primary.highlight, 0.4)}`,
    icon: <Icon name="close" color={palette.white} />,
    iconBackground: palette.primary.main,
    countDownBackground: palette.primary.main,
    titleColor: palette.primary.deepDark,
    contentColor: palette.primary.main,
    closeColor: palette.primary.light,
  },
  success: {
    background: palette.success.highlight,
    boxShadow: `0px 0px 16px ${hexToRgba(palette.success.highlight, 0.4)}`,
    icon: (
      <Icon
        name="checkbox_circle"
        type="fill"
        size="lg"
        color={palette.success.main}
      />
    ),
    iconBackground: palette.success.highlight,
    countDownBackground: palette.success.main,
    titleColor: palette.success.deepDark,
    contentColor: palette.success.main,
    closeColor: palette.success.light,
  },
  warning: {
    background: palette.warning.highlight,
    boxShadow: `0px 0px 16px ${hexToRgba(palette.warning.highlight, 0.4)}`,
    icon: <Icon name="close" color={palette.white} />,
    iconBackground: palette.warning.main,
    countDownBackground: palette.warning.main,
    titleColor: palette.warning.deepDark,
    contentColor: palette.warning.main,
    closeColor: palette.warning.light,
  },
  error: {
    background: palette.danger.highlight,
    boxShadow: `0px 0px 16px ${hexToRgba(palette.danger.highlight, 0.4)}`,
    icon: (
      <Icon name="alart" type="fill" size="lg" color={palette.danger.main} />
    ),
    iconBackground: palette.danger.highlight,
    countDownBackground: palette.danger.main,
    titleColor: palette.danger.deepDark,
    contentColor: palette.danger.main,
    closeColor: palette.danger.light,
  },
});

type TranslateMapKey = "right" | "left" | "bottom" | "top";
const translateMap = {
  right: "translate3d(120%, 0, 0)",
  left: "translate3d(-120%, 0, 0)",
  bottom: "translate3d(0, 120%, 0)",
  top: "translate3d(0, -120%, 0)",
};
function getTranslate(placement: Placement) {
  const pos = placement.split("-");
  const relevantPlacement = pos[1] === "center" ? pos[0] : pos[1];

  return translateMap[relevantPlacement as TranslateMapKey];
}

const toastStates = (placement: Placement) => ({
  entering: { transform: `${getTranslate(placement)}`, opacity: 0 },
  entered: { transform: "translate3d(0,0,0)", opacity: 1 },
  exiting: { transform: "scale(0.66)", opacity: 0 },
  exited: { transform: "scale(0.66)", opacity: 0 },
});

type Props = {} & ToastProps;

const DefaultToast: React.FunctionComponent<Props> = ({
  appearance,
  placement,
  transitionDuration,
  transitionState,
  onDismiss,
  children,
  autoDismiss,
  autoDismissTimeout,
  isRunning,
}) => {
  const theme = useTheme();
  const toastStyle = getToastStyles(theme)[appearance];
  const [height, setHeight] = React.useState<number | "auto">("auto");
  const elementRef = React.useRef<HTMLDivElement>(null);

  const handleClose = () => {
    onDismiss();
  };

  React.useEffect(() => {
    if (transitionState === "entered") {
      const el = elementRef.current;
      if (el) {
        setHeight(el.offsetHeight + theme.spacing);
      }
    }
    if (transitionState === "exiting") {
      setHeight(0);
    }
  }, [transitionState, theme.spacing]);
  return (
    <Styled.Wrapper
      ref={elementRef}
      style={{ height }}
      transitionDuration={transitionDuration}
    >
      <Styled.Container
        background={toastStyle.background}
        boxShadow={toastStyle.boxShadow}
        transitionDuration={transitionDuration}
        {...toastStates(placement)[transitionState]}
      >
        <Flex
          display="flex"
          flex={1}
          alignItems="center"
          justifyContent="space-between"
        >
          <Flex display="flex" alignItems="center">
            <Styled.IconContainer background={toastStyle.iconBackground}>
              {toastStyle.icon}
            </Styled.IconContainer>
            <Typography color={toastStyle.titleColor} weight="bold">
              {children}
            </Typography>
          </Flex>
          <Styled.CloseIconContainer onClick={handleClose}>
            <Icon name="close" color={toastStyle.closeColor} />
          </Styled.CloseIconContainer>
        </Flex>
        <Styled.Countdown
          opacity={autoDismiss ? 1 : 0}
          autoDismissTimeout={autoDismissTimeout}
          isRunning={isRunning}
          backgroundColor={toastStyle.countDownBackground}
        />
      </Styled.Container>
    </Styled.Wrapper>
  );
};

export default DefaultToast;
