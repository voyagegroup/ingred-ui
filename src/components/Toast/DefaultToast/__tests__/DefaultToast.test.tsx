import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import DefaultToast from "..";
import { renderWithThemeProvider } from "../../../../utils/renderWithThemeProvider";
import { ToastProps } from "../../../../lib/react-toast-notification/src";

const toastProps: Omit<ToastProps, "children"> = {
  appearance: "info",
  autoDismiss: false,
  autoDismissTimeout: 0,
  isRunning: true,
  onDismiss: () => {},
  onMouseEnter: () => {},
  onMouseLeave: () => {},
  placement: "top-center",
  transitionDuration: 0,
  transitionState: "entered",
};

describe("DefaultToast component testing", () => {
  afterEach(cleanup);

  test("DefaultToast info", () => {
    const { asFragment } = renderWithThemeProvider(
      <DefaultToast {...toastProps} appearance="info">
        hoge
      </DefaultToast>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("DefaultToast success", () => {
    const { asFragment } = renderWithThemeProvider(
      <DefaultToast {...toastProps} appearance="success">
        hoge
      </DefaultToast>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("DefaultToast warning", () => {
    const { asFragment } = renderWithThemeProvider(
      <DefaultToast {...toastProps} appearance="warning">
        hoge
      </DefaultToast>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("DefaultToast error", () => {
    const { asFragment } = renderWithThemeProvider(
      <DefaultToast {...toastProps} appearance="error">
        hoge
      </DefaultToast>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
