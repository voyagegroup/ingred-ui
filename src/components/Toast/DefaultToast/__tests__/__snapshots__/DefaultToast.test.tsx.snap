// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultToast component testing DefaultToast error 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL Monau"
    style="height: 8px;"
  >
    <div
      class="sc-gsDKAQ foyYFI"
      opacity="1"
      transform="translate3d(0,0,0)"
    >
      <div
        class="sc-jRQBWg cejAfR"
        display="flex"
      >
        <div
          class="sc-jRQBWg ghqqWG"
          display="flex"
        >
          <div
            class="sc-dkPtRN jkFIpl"
          >
            <div
              class="sc-gKclnd kTrEUX"
              size="24"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0,0H24V24H0Z"
                  fill="none"
                />
                <path
                  d="M12.866,3l9.526,16.5a1,1,0,0,1-.866,1.5H2.474a1,1,0,0,1-.866-1.5L11.134,3a1,1,0,0,1,1.732,0ZM11,16v2h2V16Zm0-7v5h2V9Z"
                  fill="#EB0A4E"
                />
              </svg>
            </div>
          </div>
          <p
            class="sc-iCfMLu iUnSIe"
            color="#A90556"
            font-size="14px"
          >
            hoge
          </p>
        </div>
        <div
          class="sc-hKwDye kjnnxs"
        >
          <div
            class="sc-gKclnd edtajv"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#F96979"
                height="23.141"
                rx="2.571"
                transform="translate(18.364 2) rotate(45)"
                width="5.142"
              />
              <rect
                fill="#F96979"
                height="23.141"
                rx="2.571"
                transform="translate(2 5.636) rotate(-45)"
                width="5.143"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-eCImPb gEmtyD"
        opacity="0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DefaultToast component testing DefaultToast info 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL Monau"
    style="height: 8px;"
  >
    <div
      class="sc-gsDKAQ hMnbFy"
      opacity="1"
      transform="translate3d(0,0,0)"
    >
      <div
        class="sc-jRQBWg cejAfR"
        display="flex"
      >
        <div
          class="sc-jRQBWg ghqqWG"
          display="flex"
        >
          <div
            class="sc-dkPtRN lnGcAY"
          >
            <div
              class="sc-gKclnd edtajv"
              size="18"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#FDFEFF"
                  height="23.141"
                  rx="2.571"
                  transform="translate(18.364 2) rotate(45)"
                  width="5.142"
                />
                <rect
                  fill="#FDFEFF"
                  height="23.141"
                  rx="2.571"
                  transform="translate(2 5.636) rotate(-45)"
                  width="5.143"
                />
              </svg>
            </div>
          </div>
          <p
            class="sc-iCfMLu fxmbOw"
            color="#054BAF"
            font-size="14px"
          >
            hoge
          </p>
        </div>
        <div
          class="sc-hKwDye kjnnxs"
        >
          <div
            class="sc-gKclnd edtajv"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#9CDDFD"
                height="23.141"
                rx="2.571"
                transform="translate(18.364 2) rotate(45)"
                width="5.142"
              />
              <rect
                fill="#9CDDFD"
                height="23.141"
                rx="2.571"
                transform="translate(2 5.636) rotate(-45)"
                width="5.143"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-eCImPb jCBjVe"
        opacity="0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DefaultToast component testing DefaultToast success 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL Monau"
    style="height: 8px;"
  >
    <div
      class="sc-gsDKAQ jFdwci"
      opacity="1"
      transform="translate3d(0,0,0)"
    >
      <div
        class="sc-jRQBWg cejAfR"
        display="flex"
      >
        <div
          class="sc-jRQBWg ghqqWG"
          display="flex"
        >
          <div
            class="sc-dkPtRN gPObHa"
          >
            <div
              class="sc-gKclnd kTrEUX"
              size="24"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0,0H24V24H0Z"
                  fill="none"
                />
                <path
                  d="M12,22A10,10,0,1,1,22,12,10,10,0,0,1,12,22Zm-1-6,7.07-7.071L16.659,7.515,11,13.172,8.174,10.343,6.76,11.757Z"
                  fill="#63C427"
                />
              </svg>
            </div>
          </div>
          <p
            class="sc-iCfMLu eoKRge"
            color="#338D13"
            font-size="14px"
          >
            hoge
          </p>
        </div>
        <div
          class="sc-hKwDye kjnnxs"
        >
          <div
            class="sc-gKclnd edtajv"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#B4ED7B"
                height="23.141"
                rx="2.571"
                transform="translate(18.364 2) rotate(45)"
                width="5.142"
              />
              <rect
                fill="#B4ED7B"
                height="23.141"
                rx="2.571"
                transform="translate(2 5.636) rotate(-45)"
                width="5.143"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-eCImPb cAOdnj"
        opacity="0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DefaultToast component testing DefaultToast warning 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL Monau"
    style="height: 8px;"
  >
    <div
      class="sc-gsDKAQ ikXnFi"
      opacity="1"
      transform="translate3d(0,0,0)"
    >
      <div
        class="sc-jRQBWg cejAfR"
        display="flex"
      >
        <div
          class="sc-jRQBWg ghqqWG"
          display="flex"
        >
          <div
            class="sc-dkPtRN euxYHz"
          >
            <div
              class="sc-gKclnd edtajv"
              size="18"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#FDFEFF"
                  height="23.141"
                  rx="2.571"
                  transform="translate(18.364 2) rotate(45)"
                  width="5.142"
                />
                <rect
                  fill="#FDFEFF"
                  height="23.141"
                  rx="2.571"
                  transform="translate(2 5.636) rotate(-45)"
                  width="5.143"
                />
              </svg>
            </div>
          </div>
          <p
            class="sc-iCfMLu gzQbZH"
            color="#785F03"
            font-size="14px"
          >
            hoge
          </p>
        </div>
        <div
          class="sc-hKwDye kjnnxs"
        >
          <div
            class="sc-gKclnd edtajv"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#FDE54E"
                height="23.141"
                rx="2.571"
                transform="translate(18.364 2) rotate(45)"
                width="5.142"
              />
              <rect
                fill="#FDE54E"
                height="23.141"
                rx="2.571"
                transform="translate(2 5.636) rotate(-45)"
                width="5.143"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-eCImPb epUjoV"
        opacity="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
